### ISSUE TITLE
floating point in equality comparison commands.c

### ISSUE THREAD

**Issue by evanmayer** [tim-balloon/TIMflight#5]
## Task

Floating point comparisons of (in)equality are used in command logic.

[`./mcp/commanding/commands.c`](https://github.com/tim-balloon/TIMflight/blob/7361f99be8985bd09c8c706b4e3d8c8f22c4e46c/mcp/commanding/commands.c#L1441):

```C
      if ((CommandData.pointing_mode.mode != P_AZEL_GOTO) ||
          (CommandData.pointing_mode.X != rvalues[0]) ||
          (CommandData.pointing_mode.Y != rvalues[1])) {
        CommandData.pointing_mode.nw = CommandData.slew_veto;
      }
```

`rvalues` being an array of `double`s.

This is bad because the result of the written code is [not guaranteed to match the intent](https://bitbashing.io/comparing-floats.html), so we should fix it wherever we see it.

## Acceptance

Static analysis of `commands.c` proves a reduction of this type of violation, which is checked by many basic rulesets.

## Details

There are so many of these (even in just this file) that it's likely to be a systemic issue throughout the code. Write a macro or function to perform a better floating point comparison and put it in a header or implementation file somewhere so it can be included and used elsewhere easily. You probably don't need to go as crazy as the author does in the link above, if you can make some reasoned assumptions about the majority of values we'll be comparing.

**Comment by evanmayer** [comment]
Closed by #14!

**Comment by evanmayer** [comment]
### SciTools Understand Codecheck static analysis

* MISRA C 2004 rule 13.3 only
* `./mcp` only

#### Before ([447b069a82101dfc165c195d66529d4665ec9297](https://github.com/tim-balloon/TIMflight/commit/447b069a82101dfc165c195d66529d4665ec9297)):

* 59 violations in 130 files
* including `commands.c`

#### After: ([447b069a82101dfc165c195d66529d4665ec9297](https://github.com/tim-balloon/TIMflight/commit/7243443c365e06655269a7c33acc1500cf316963)):

* 37 violations in 130 files
* no violations in `commands.c`

> I should note that upon looking at MISRA C 2012, this rule (MISRA C 2004 rule 13.3) has been revised to a more generic Directive 1.1: "Any implementation-defined behaviour on which the output of
the program depends shall be documented and understood". Still a good check for us though, IMO.

This violation is reduced and we now have a tool for fixing it in other places we see it. Therefore, resolved by #14.
